### PR TITLE
gateway: Adding additional functionality and some missing files

### PR DESCRIPTION
**Pull request by djgalloway**

### CODE REVIEW

File: roles/gateway/tasks/users.yml [hunk 1]
```diff
@@ -11,3 +11,11 @@
     owner: root
     group: root
     mode: 0644
+
+- name: Upload auth-openvpn script
+  template:
```
Comment (dmick) [inline roles/gateway/tasks/users.yml]: This doesn't need to be a template, right?  there's no substitutions I see?

File: roles/gateway/tasks/main.yml [hunk 2]
```diff
@@ -19,6 +19,7 @@
 # Manage OpenVPN users list using secrets repo
 - include: users.yml
   tags:
+    - always
```
Comment (dmick) [inline roles/gateway/tasks/main.yml]: Why always?  and if we need it, doesn't this make -users pointless?

File: roles/gateway/README.rst [hunk 3]
```diff
@@ -23,11 +23,15 @@ A list of packages to install that is specific to the role.  The list is defined
 
     packages: []
 
+A unique name to give to your OpenVPN service.  This name is used to organize configuration files and start/stop the service.  Defined in the secrets repo::
+
+    openvpn_server_name: []
```
Comment (dmick) [inline roles/gateway/README.rst]: If this isn't given a default, is there something which will check for it being set and error if not?

File: roles/gateway/tasks/main.yml (hunk 2 above)
Comment (djgalloway) [inline roles/gateway/tasks/main.yml]: I feel like I had a better explanation for this yesterday but now all I can think of is "why not?"

One can still call the users tag directly and only run those tasks so the tag isn't useless.

File: roles/gateway/README.rst (hunk 3 above)
Comment (djgalloway) [inline roles/gateway/README.rst]: Good point.  Added https://github.com/ceph/ceph-cm-ansible/pull/216/commits/1cf6fd5b68de0f5c21c7f1367b84d2c94811b5fe

File: roles/gateway/tasks/main.yml (hunk 2 above)
Comment (dmick) [inline roles/gateway/tasks/main.yml]: as to "why not": what if I want to set up an openvpn server but not set up the users file?

File: roles/gateway/defaults/main.yml [hunk 4]
```diff
@@ -5,3 +5,7 @@
 secrets_repo:
   name: UNDEFINED
   url: null
+
+openvpn_server_name: "server"
```
Comment (dmick) [inline roles/gateway/defaults/main.yml]: I know you don't need quotes here; not sure if they do any damage, but it's not the prevailing style, so, should remove them

File: roles/gateway/tasks/main.yml (hunk 2 above)
Comment (djgalloway) [inline roles/gateway/tasks/main.yml]: `--skip-tags="users"` is a thing, but point taken.  Removed.

**Pull request merged**